File: LeetCode/LeetCode Practice Problems/1704- Determine if String Halves Are Alike.cpp
class Solution {
public:
    bool halvesAreAlike(string s) {
        int cnt = 0;
        int i=0,j=s.length()-1;
        
        while(i < j) {
            s[i] = tolower(s[i]);
            s[j] = tolower(s[j]);
            
            if(s[i] == 'a' || s[i] == 'e' || s[i] == 'i' || s[i] == 'o' || s[i] == 'u') cnt++;
            if(s[j] == 'a' || s[j] == 'e' || s[j] == 'i' || s[j] == 'o' || s[j] == 'u') cnt--;
            i++;
            j--;
        }
        
        
        return cnt == 0;

    }
};

File: LeetCode/LeetCode Practice Problems/1704-Determine if String Halves Are Alike.java
class Solution {
    public boolean halvesAreAlike(String s) {
        String vowels = "aeiouAEIOU";
        int cnt1 = 0;
        int cnt2 = 0;
        int n = s.length();
        
        for(int i=0;i<n;i++) {
            char ch = s.charAt(i);
            if(vowels.contains(ch + "")) {
                if(i < n/2) cnt1++;
                else cnt2++;
            }
        }
        
        return cnt1 == cnt2;
    }
}
